### PR TITLE
fix: upgrade k0s version and fix registry upgrade with helm upgrade --force=false

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

Upgrades k0s to fix the following error:

The registry upgrades fail with the following error as you can see from e2e TestSingleNodeAirgapUpgradeFromEC18.

```
Aug 15 18:44:57 ethanm-ec-1 k0s[1205575]: time="2024-08-15 18:44:57" level=error msg="Reconciler error" Chart="{k0s-addon-chart-docker-registry kube-system}" component=extensions_controller controller=chart controllerGroup=helm.k0sproject.io controllerKind=Chart error="can't update or install chart: can't reconcile upgrade for \"k0s-addon-chart-docker-registry\": can't upgrade loadedChart `docker-registry`: an error occurred while rolling back the release. original upgrade error: failed to replace object: PersistentVolumeClaim \"registry\" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests and volumeAttributesClassName for bound claims\n  core.PersistentVolumeClaimSpec{\n  \tAccessModes:      {\"ReadWriteOnce\"},\n  \tSelector:         nil,\n  \tResources:        {Requests: {s\"storage\": {i: {...}, s: \"10Gi\", Format: \"BinarySI\"}}},\n- \tVolumeName:       \"pvc-443acec3-b7d1-4bb0-a1fe-bc717c0077e3\",\n+ \tVolumeName:       \"\",\n  \tStorageClassName: &\"openebs-hostpath\",\n  \tVolumeMode:       &\"Filesystem\",\n  \t... // 3 identical fields\n  }\n: failed to replace object: PersistentVolumeClaim \"registry\" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests and volumeAttributesClassName for bound claims\n  core.PersistentVolumeClaimSpec{\n  \tAccessModes:      {\"ReadWriteOnce\"},\n  \tSelector:         nil,\n  \tResources:        {Requests: {s\"storage\": {i: {...}, s: \"10Gi\", Format: \"BinarySI\"}}},\n- \tVolumeName:       \"pvc-443acec3-b7d1-4bb0-a1fe-bc717c0077e3\",\n+ \tVolumeName:       \"\",\n  \tStorageClassName: &\"openebs-hostpath\",\n  \tVolumeMode:       &\"Filesystem\",\n  \t... // 3 identical fields\n  }\n" name=k0s-addon-chart-docker-registry namespace=kube-system reconcileID="\"6ba101f6-a2a3-4421-80b8-295179661b1f\""
```

This happens because k0s uses the upgrade --force flag

https://github.com/k0sproject/k0s/blob/v1.29.7%2Bk0s.0/pkg/helm/helm.go#L283

See relevant helm issue here

https://github.com/helm/helm/issues/11729#issuecomment-1401989311

There is a race condition which i believe to be a bug in k0s that sometimes prevents this error from showing up in the chart object and thus in the ui.

Detailed analysis here

https://docs.google.com/document/d/17ko0az-JWygal83j6LJ4XGZuLMEVD2F4kWgnauSwv3U/edit

#### Does this PR require a release note?

```release-note
Fixes an issue that can cause the Docker registry chart to to fail to upgrade with an invalid spec error.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,6 +25,15 @@ LOCAL_ARTIFACT_MIRROR_IMAGE ?= proxy.replicated.com/anonymous/replicated/embedde
 METADATA_K0S_BINARY_URL_OVERRIDE =
 METADATA_KOTS_BINARY_URL_OVERRIDE =
 METADATA_OPERATOR_BINARY_URL_OVERRIDE =
+
+ifeq ($(ARCH),amd64)
+ifeq ($(K0S_VERSION),v1.29.9+k0s.0)
+K0S_BINARY_SOURCE_OVERRIDE = https://embedded-cluster-assets.s3.amazonaws.com/k0s-v1.29.9%2Bk0s.0-amd64
```
Comment (emosbaugh) [inline Makefile:31]: contains this patch

https://github.com/k0sproject/k0s/compare/main...emosbaugh:k0s:fix-forceUpgrade-nil-default?expand=1

Comment (laverya) [reply Makefile:31]: do we need this if my PR gets merged?

Comment (emosbaugh) [reply Makefile:31]: What would change this about your pr?

**Review by sgalsaleh — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (laverya) [reply Makefile:31]: Well, with the new upgrade path (job in cluster, I mean), we don't need this patch
however, that won't fix the issue when upgrading from pre-1.8

File: operator/pkg/charts/charts.go [hunk 2]
```diff
@@ -126,6 +128,19 @@ func mergeHelmConfigs(ctx context.Context, meta *ectypes.ReleaseMetadata, in *cl
...
 func updateInfraChartsFromInstall(in *v1beta1.Installation, clusterConfig *k0sv1beta1.ClusterConfig, charts []v1beta1.Chart) ([]v1beta1.Chart, error) {
 	for i, chart := range charts {
+		ecCharts := []string{
+			"admin-console",
+			"docker-registry",
+			"embedded-cluster-operator",
+			"openebs",
+			"seaweedfs",
+			"velero",
+		}
```
[the remark this answers is not in the document]
Comment (laverya) [reply operator/pkg/charts/charts.go:138]: nope, I was wrong
ignore me

Do we have a way for vendors to specify that they want to not force install their charts, then?

Or will that take a kinds PR?

**Review by laverya — APPROVED** [review]

File: operator/pkg/charts/charts.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (emosbaugh) [reply operator/pkg/charts/charts.go:138]: Kinds has been updated

https://github.com/replicatedhq/embedded-cluster/pull/1261/files#diff-997a5ee12458e3973a7074aca8bb284277ede51e07f8adb06fcb65467cec0072R98

**Pull request merged**